_posts/2024-11-18-datafusion-fastest-single-node-parquet-clickbench.md

```diff
@@ -0,0 +1,347 @@
+* **DataFusion**:  [Xiangpeng Hao](https://github.com/XiangpengHao), again charted the initial path and [Alex Huang](https://github.com/Weijun-H), [Dharan Aditya](https://github.com/dharanad) [Lordworms](https://github.com/Lordworms), [Jax Liu](https://github.com/goldmedal),  [wiedld](https://github.com/wiedld), [Tai Le Manh](https://github.com/tlm365), [yi wang](https://github.com/my-vegetable-has-exploded), [doupache](https://github.com/doupache), [Jay Zhan](https://github.com/jayzhan211) , [Xin Li](https://github.com/xinlifoobar)  and [Kaifeng Zheng](https://github.com/Kev1n8) made it real.
+* **DataFusion String Function Migration**:  [Trent Hauck](https://github.com/tshauck) organized the effort and set the patterns, [Jax Liu](https://github.com/goldmedal) made a clever testing framework, and [Austin Liu](https://github.com/austin362667), [Dmitrii Bu](https://github.com/demetribu), [Tai Le Manh](https://github.com/tlm365), [Chojan Shang](https://github.com/PsiACE), [WeblWabl](https://github.com/devanbenz), [Lordworms](https://github.com/Lordworms), [iamthinh](https://github.com/thinh2), [Bruce Ritchie](https://github.com/Omega359), [Kaifeng Zheng](https://github.com/Kev1n8), and [Xin Li](https://github.com/xinlifoobar) bashed out the conversions.
+
+
+## Parquet
+
+Part DataFusion's speed in ClickBench is reading Parquet files (really) quickly,
+which reflects invested effort in the Parquet reading system (see [Querying
+Parquet with Millisecond Latency]
+
+The [DataFusion ParquetExec] (built on the [Rust Parquet Implementation]) is now the most
+sophisticated open source Parquet reader I know of. It has every optimization we
+can think of for reading Parquet, including projection pushdown, predicate
+pushdown (row group metadata, page index, and bloom filters), limit pushdown,
+parallel reading, interleaved I/O, and late materialized filtering (coming soon ™️
+by default). Some recent work from [June](https://github.com/itsjunetime)
+[recently unblocked a remaining hurdle] for enabling late materialized
+filtering, and conveniently [Xiangpeng Hao](https://github.com/XiangpengHao) is
+working on the [final piece] (no pressure😅)
+
+[recently unblocked a remaining hurdle]: https://github.com/apache/datafusion/pull/12135
+[final piece]: https://github.com/apache/arrow-datafusion/issues/3463
+
+[Querying Parquet with Millisecond Latency]: https://www.influxdata.com/blog/querying-parquet-millisecond-latency/
+[DataFusion ParquetExec]: https://docs.rs/datafusion/latest/datafusion/datasource/physical_plan/parquet/struct.ParquetExec.html
+[Rust Parquet Implementation]: https://crates.io/crates/parquet
+
+## Skipping Partial Aggregation When Doesn't Help
```

Comment:
**Skipping Partial Aggregation When Doesn't Help**

Is there a missing word in this title?

Reply:
```suggestion
## Skipping Partial Aggregation When It Doesn't Help
```